src/document/code.rs

```diff
@@ -266,20 +208,18 @@ impl Line {
                     writeln!(
                         result,
                         "{}",
-                        block
-                            .compile_with(code_blocks, scope, settings)
-                            .map(|code| {
-                                code.split('\n')
-                                    .map(|line| {
-                                        if blank_lines && line.trim().is_empty() {
-                                            "".to_string()
-                                        } else {
-                                            format!("{}{}", self.indent, line)
-                                        }
-                                    })
-                                    .collect::<Vec<_>>()
-                                    .join("\n")
-                            })?,
+                        block.compile_with(code_blocks, settings).map(|code| {
+                            code.split('\n')
```

Comment:
Not sure if you want to touch this now or just earmark it for later, but
* `some_str.split('\n')` is probably better written as `some_str.lines()` using [`str::lines`](https://doc.rust-lang.org/stable/std/primitive.str.html#method.lines)
* this might benefit from using the non-collecting `JoinExt::join` function or even manually writing everything into a single `String` to avoid the intermediate `format!` calls for the indentation

Reply:
Done. Besides avoiding collect and format, it also makes the code much more concise:

```rust
let code = block.compile_with(code_blocks, settings)?;
for line in code.lines() {
    if blank_lines && line.trim().is_empty() {
        writeln!(result).unwrap();
    } else {
        write!(result, "{}", self.indent).unwrap();
        writeln!(result, "{}", line).unwrap();
    }
}
```

Maybe an alternative to `line.trim().is_empty()` would be helpful, but is does not allocate, so it is probably ok.